
* Put 3DES off as insecure.
* Always use "linux" terminal type. This resolves some
  strange key combos.
* Set standard colors and use a more legilible font size.

---
 settings.c        |   49 +++++++++++++++++++++++++++++++++----------------
 unix/unix.h       |    6 +-----
 windows/windefs.c |    2 +-
 3 files changed, 35 insertions(+), 22 deletions(-)

Index: putty-0.72/settings.c
===================================================================
--- putty-0.72.orig/settings.c
+++ putty-0.72/settings.c
@@ -17,8 +17,8 @@
 static const struct keyvalwhere ciphernames[] = {
     { "aes",        CIPHER_AES,             -1, -1 },
     { "chacha20",   CIPHER_CHACHA20,        CIPHER_AES, +1 },
-    { "3des",       CIPHER_3DES,            -1, -1 },
     { "WARN",       CIPHER_WARN,            -1, -1 },
+    { "3des",       CIPHER_3DES,            -1, -1 },
     { "des",        CIPHER_DES,             -1, -1 },
     { "blowfish",   CIPHER_BLOWFISH,        -1, -1 },
     { "arcfour",    CIPHER_ARCFOUR,         -1, -1 },
@@ -847,7 +847,7 @@ void load_open_settings(settings_r *sess
     }
     gppb(sesskey, "TCPNoDelay", true, conf, CONF_tcp_nodelay);
     gppb(sesskey, "TCPKeepalives", false, conf, CONF_tcp_keepalives);
-    gpps(sesskey, "TerminalType", "xterm", conf, CONF_termtype);
+    gpps(sesskey, "TerminalType", "linux", conf, CONF_termtype);
     gpps(sesskey, "TerminalSpeed", "38400,38400", conf, CONF_termspeed);
     if (gppmap(sesskey, "TerminalModes", conf, CONF_ttymodes)) {
 	/*
@@ -1029,10 +1029,10 @@ void load_open_settings(settings_r *sess
     gppb(sesskey, "PassiveTelnet", false, conf, CONF_passive_telnet);
     gppb(sesskey, "BackspaceIsDelete", true, conf, CONF_bksp_is_delete);
     gppb(sesskey, "RXVTHomeEnd", false, conf, CONF_rxvt_homeend);
-    gppi(sesskey, "LinuxFunctionKeys", 0, conf, CONF_funky_type);
+    gppi(sesskey, "LinuxFunctionKeys", true, conf, CONF_funky_type);
     gppb(sesskey, "NoApplicationKeys", false, conf, CONF_no_applic_k);
     gppb(sesskey, "NoApplicationCursors", false, conf, CONF_no_applic_c);
-    gppb(sesskey, "NoMouseReporting", false, conf, CONF_no_mouse_rep);
+    gppb(sesskey, "NoMouseReporting", true, conf, CONF_no_mouse_rep);
     gppb(sesskey, "NoRemoteResize", false, conf, CONF_no_remote_resize);
     gppb(sesskey, "NoAltScreen", false, conf, CONF_no_alt_screen);
     gppb(sesskey, "NoRemoteWinTitle", false, conf, CONF_no_remote_wintitle);
@@ -1054,9 +1054,9 @@ void load_open_settings(settings_r *sess
     gppb(sesskey, "ApplicationKeypad", false, conf, CONF_app_keypad);
     gppb(sesskey, "NetHackKeypad", false, conf, CONF_nethack_keypad);
     gppb(sesskey, "AltF4", true, conf, CONF_alt_f4);
-    gppb(sesskey, "AltSpace", false, conf, CONF_alt_space);
+    gppb(sesskey, "AltSpace", true, conf, CONF_alt_space);
     gppb(sesskey, "AltOnly", false, conf, CONF_alt_only);
-    gppb(sesskey, "ComposeKey", false, conf, CONF_compose_key);
+    gppb(sesskey, "ComposeKey", true, conf, CONF_compose_key);
     gppb(sesskey, "CtrlAltKeys", true, conf, CONF_ctrlaltkeys);
 #ifdef OSX_META_KEY_CONFIG
     gppb(sesskey, "OSXOptionMeta", true, conf, CONF_osx_option_meta);
@@ -1068,12 +1068,12 @@ void load_open_settings(settings_r *sess
     gppi(sesskey, "LocalEdit", AUTO, conf, CONF_localedit);
     gpps(sesskey, "Answerback", "PuTTY", conf, CONF_answerback);
     gppb(sesskey, "AlwaysOnTop", false, conf, CONF_alwaysontop);
-    gppb(sesskey, "FullScreenOnAltEnter", false,
+    gppb(sesskey, "FullScreenOnAltEnter", true,
          conf, CONF_fullscreenonaltenter);
     gppb(sesskey, "HideMousePtr", false, conf, CONF_hide_mouseptr);
     gppb(sesskey, "SunkenEdge", false, conf, CONF_sunken_edge);
     gppi(sesskey, "WindowBorder", 1, conf, CONF_window_border);
-    gppi(sesskey, "CurType", 0, conf, CONF_cursor_type);
+    gppi(sesskey, "CurType", 1, conf, CONF_cursor_type);
     gppb(sesskey, "BlinkCur", false, conf, CONF_blink_cur);
     /* pedantic compiler tells me I can't use conf, CONF_beep as an int * :-) */
     gppi(sesskey, "Beep", 1, conf, CONF_beep);
@@ -1108,10 +1108,10 @@ void load_open_settings(settings_r *sess
     gppb(sesskey, "CRImpliesLF", false, conf, CONF_crhaslf);
     gppb(sesskey, "DisableArabicShaping", false, conf, CONF_no_arabicshaping);
     gppb(sesskey, "DisableBidi", false, conf, CONF_no_bidi);
-    gppb(sesskey, "WinNameAlways", true, conf, CONF_win_name_always);
+    gppb(sesskey, "WinNameAlways", false, conf, CONF_win_name_always);
     gpps(sesskey, "WinTitle", "", conf, CONF_wintitle);
     gppi(sesskey, "TermWidth", 80, conf, CONF_width);
-    gppi(sesskey, "TermHeight", 24, conf, CONF_height);
+    gppi(sesskey, "TermHeight", 25, conf, CONF_height);
     gppfont(sesskey, "Font", conf, CONF_font);
     gppi(sesskey, "FontQuality", FQ_DEFAULT, conf, CONF_font_quality);
     gppi(sesskey, "FontVTMode", VT_UNICODE, conf, CONF_vtmode);
@@ -1124,11 +1124,28 @@ void load_open_settings(settings_r *sess
 
     for (i = 0; i < 22; i++) {
 	static const char *const defaults[] = {
-	    "187,187,187", "255,255,255", "0,0,0", "85,85,85", "0,0,0",
-	    "0,255,0", "0,0,0", "85,85,85", "187,0,0", "255,85,85",
-	    "0,187,0", "85,255,85", "187,187,0", "255,255,85", "0,0,187",
-	    "85,85,255", "187,0,187", "255,85,255", "0,187,187",
-	    "85,255,255", "187,187,187", "255,255,255"
+            "170,170,170", /* default foreground */
+            "255,255,255", /* default bold foreground */
+            "0,0,0",       /* default background */
+            "85,85,85",    /* default bold background */
+            "0,0,0",       /* cursor text */
+            "192,192,224", /* cursor color */
+            "0,0,0",       /* black */
+            "85,85,85",    /* black bold */
+            "170,0,0",     /* red */
+            "255,0,0",     /* red bold */
+            "0,170,0",     /* green */
+            "0,255,0",     /* green bold */
+            "170,85,0",    /* brown */
+            "255,255,0",   /* yellow bold */
+            "0,0,170",     /* blue */
+            "0,0,255",     /* blue bold */
+            "170,0,170",   /* magenta */
+            "255,0,255",   /* magenta bold */
+            "0,170,170",   /* cyan */
+            "0,255,255",   /* cyan bold */
+            "170,170,170", /* white */
+            "255,255,255", /* white bold */
 	};
 	char buf[20], *buf2;
 	int c0, c1, c2;
@@ -1186,7 +1203,7 @@ void load_open_settings(settings_r *sess
      * The empty default for LineCodePage will be converted later
      * into a plausible default for the locale.
      */
-    gpps(sesskey, "LineCodePage", "", conf, CONF_line_codepage);
+    gpps(sesskey, "LineCodePage", "UTF-8", conf, CONF_line_codepage);
     gppb(sesskey, "CJKAmbigWide", false, conf, CONF_cjk_ambig_wide);
     gppb(sesskey, "UTF8Override", true, conf, CONF_utf8_override);
     gpps(sesskey, "Printer", "", conf, CONF_printer);
Index: putty-0.72/unix/unix.h
===================================================================
--- putty-0.72.orig/unix/unix.h
+++ putty-0.72/unix/unix.h
@@ -400,11 +400,7 @@ Socket *make_fd_socket(int infd, int out
 /*
  * Default font setting, which can vary depending on NOT_X_WINDOWS.
  */
-#ifdef NOT_X_WINDOWS
-#define DEFAULT_GTK_FONT "client:Monospace 12"
-#else
-#define DEFAULT_GTK_FONT "server:fixed"
-#endif
+#define DEFAULT_GTK_FONT "client:Monospace 14"
 
 /*
  * uxpty.c.
Index: putty-0.72/windows/windefs.c
===================================================================
--- putty-0.72.orig/windows/windefs.c
+++ putty-0.72/windows/windefs.c
@@ -9,7 +9,7 @@
 FontSpec *platform_default_fontspec(const char *name)
 {
     if (!strcmp(name, "Font"))
-        return fontspec_new("Courier New", false, 10, ANSI_CHARSET);
+        return fontspec_new("Consolas", false, 14, ANSI_CHARSET);
     else
         return fontspec_new("", false, 0, 0);
 }
